action/action.yml

```diff
@@ -26,13 +26,12 @@ runs:
       run: |
         VERSION=${{ inputs.version }}
 
-        TOKEN=${{ inputs.token }}
-        if [[ -z "$TOKEN" ]]; then
-          TOKEN=${{ github.token }}
-        fi
-
         if [[ -z "$VERSION" ]] || [[ "$VERSION" = "latest" ]]; then
-          VERSION=$(curl -fsSL -H "Authorization: token ${TOKEN}" https://api.github.com/repos/fluxcd/flux2/releases/latest | grep tag_name | cut -d '"' -f 4)
+          if [[ "${{ inputs.token }}" != '' ]]; then
```

Comment:
Interpolating values from GitHub Actions in shell scripts is a bad practice. Can you please pass the token through an environment variable to the shell script? Like this:

```yaml
      env:
        TOKEN: ${{ inputs.token }}
```

No need to fix the other inputs now, but it would be good to fix the one you are touching in this PR :pray: